graph/patterns/alternate-key.md

```diff
@@ -0,0 +1,135 @@
+2. If the workload does not have a list of alternate keys, it can still query the data source, but if the result yields more than a record, it SHOULD return `422` and encourage the user to use `$filter` rather than returning the first result of the query
+
+## Example
+
+---
+
+The same user identified via the alternate key SSN, the canonical (primary) key ID using the non-canonical long form with specified key property name, and the canonical short form without key property name
+
+Declare `mail` and `ssn` as alternate keys on an entity:
+
+```xml
+<EntityType Name="User">
+   <Key>
+     <PropertyRef Name="id" />
+   </Key>
+   <Property Name="id" Type="Edm.Int32" />
+
+   <Property Name="mail" Type="Edm.String" />
+   <Property Name="ssn" Type="Edm.String" />
+   <Annotation Term="Keys.AlternateKeys">
+      <Collection>
+         <Record>
+            <PropertyValue Property="Key">
+               <Collection>
+                  <Record>
+                     <PropertyValue Property="Name" PropertyPath="mail" />
+                  </Record>
+               </Collection>
+            </PropertyValue>
+         </Record>
+         <Record>
+            <PropertyValue Property="Key">
+               <Collection>
+                  <Record>
+                     <PropertyValue Property="Name" PropertyPath="ssn" />
+                  </Record>
+               </Collection>
+            </PropertyValue>
+         </Record>
+      </Collection>
+   </Annotation>
+</EntityType>
+```
+
+1. Get a specific resource through `$filter`:
+
+```http
+GET https://graph.microsoft.com/v1.0/users/?$filter=(ssn eq '123-45-6789')
+```
+
+```json
+{
+  "value": [
+    {
+      "givenName": "Bob",
+      "jobTitle": "Retail Manager",
+      "mail": "bob@contoso.com",
+      "mobilePhone": "+1 425 555 0109",
+      "officeLocation": "18/2111",
+      "preferredLanguage": "en-US",
+      "surname": "Vance",
+      "userPrincipalName": "bob@contoso.com",
+      "id": "1a89ade6-9f59-4fea-a139-23f84e3aef66"
+    }
+  ]
+}
+```
+
+2. Get a specific resource through its primary key, and then through the two alternate keys:
+
+```http
+GET https://graph.microsoft.com/v1.0/users/1a89ade6-9f59-4fea-a139-23f84e3aef66
+GET https://graph.microsoft.com/v1.0/users(ssn='123-45-6789')
+GET https://graph.microsoft.com/v1.0/users(email='bob@contoso.com')
+```
+
+All of the 3 will yield the sare response:
+
+```json
+{
+  "givenName": "Bob",
+  "jobTitle": "Retail Manager",
+  "mail": "bob@contoso.com",
+  "mobilePhone": "+1 425 555 0109",
+  "officeLocation": "18/2111",
+  "preferredLanguage": "en-US",
+  "surname": "Vance",
+  "userPrincipalName": "bob@contoso.com",
+  "id": "1a89ade6-9f59-4fea-a139-23f84e3aef66"
+}
+```
+
+3. Requesting a resource through an alternate key which yields more than a result
+
+```http
+GET https://graph.microsoft.com/v1.0/users(name='Bob')
+
+400 Bad Request
+```
```

Comment:
Should we add an example of the case where multiple properties are required to form an alternative key e.g. https://graph.microsoft.com/v1.0/users(email='bob@contoso.com',ssn='1234')